sdcclient/_scanning.py

```diff
@@ -0,0 +1,535 @@
+
+
+    def activate_subscription(self, subscription_type, subscription_key):
+        '''**Description**
+            Activate a subscription
+
+        **Arguments**
+            - subscription_type: Type of subscription. Valid options:
+                - 'tag_update': Receive notification when new image is pushed
+                - 'policy_eval': Receive notification when image policy status changes
+                - 'vuln_update': Receive notification when vulnerabilities are added, removed or modified
+            - subscription_key: Fully qualified name of tag to subscribe to. Eg. docker.io/library/alpine:latest
+        '''
+        return self._update_subscription(subscription_type, subscription_key, True)
+
+    def deactivate_subscription(self, subscription_type, subscription_key):
+        '''**Description**
+            Deactivate a subscription
+
+        **Arguments**
+            - subscription_type: Type of subscription. Valid options:
+                - 'tag_update': Receive notification when new image is pushed
+                - 'policy_eval': Receive notification when image policy status changes
+                - 'vuln_update': Receive notification when vulnerabilities are added, removed or modified
+            - subscription_key: Fully qualified name of tag to subscribe to. Eg. docker.io/library/alpine:latest
+        '''
+        return self._update_subscription(subscription_type, subscription_key, False)
+
+    def _update_subscription(self, subscription_type, subscription_key, activate):
+        hashstr = '+'.join([self.token, subscription_key, subscription_type]).encode('utf-8')
+        subscription_id = hashlib.md5(hashstr).hexdigest()
+        url = self.url + "/api/scanning/v1/anchore/subscriptions/" + subscription_id
+        payload = {'active': activate, 'subscription_key': subscription_key, 'subscription_type': subscription_type}
+
+        res = requests.put(url, data=json.dumps(payload), headers=self.hdrs, verify=self.ssl_verify)
+        if not self._checkResponse(res):
+            return [False, self.lasterr]
+
+        return [True, res.json()]
+
+    def list_subscription(self):
+        '''**Description**
+            List all subscriptions
+
+        **Arguments**
+            - None
+
+        **Success Return Value**
+            A JSON object representing the list of subscriptions.
+        '''
+        url = self.url + "/api/scanning/v1/anchore/subscriptions"
+        res = requests.get(url, headers=self.hdrs, verify=self.ssl_verify)
+        if not self._checkResponse(res):
+            return [False, self.lasterr]
+
+        return [True, res.json()]
+
+    def _discover_inputimage_format(self, input_string):
+        itype = None
+
+        if re.match("^sha256:[0-9a-fA-F]{64}", input_string):
+            itype = 'imageDigest'
+        elif re.match("[0-9a-fA-F]{64}", input_string):
+            itype = 'imageid'
+        else:
+            itype = 'tag'
+
+        return(itype)
```

Comment:
We don't need the `(`, `)` here. But existing code has them all around, so :see_no_evil: 

We need a refactor there.